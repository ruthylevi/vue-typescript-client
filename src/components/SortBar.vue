<template>
  <v-toolbar dense>
    <v-menu>
      <template v-slot:activator="{ on }">
        <v-btn color="primary" dark v-on="on">{{sort}}</v-btn>
      </template>
      <v-list>
        <v-list-item
          v-for="(item, index) in dropdown_sort"
          :key="index"
          @click="sortbyUpdate(item.title)"
        >
          <v-list-item-title>{{ item.title }}</v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
      <v-divider vertical></v-divider>
      <v-btn @click="orderUpdate">
        <v-icon>{{order=="descending" ? 'mdi-sort-descending' : 'mdi-sort-ascending'}}</v-icon>
      </v-btn>
      <v-spacer></v-spacer>
      <v-spacer></v-spacer>
      <v-spacer></v-spacer>

      <div class="mx-4"></div>
  </v-toolbar>
</template>

<script>
export default {
  data() {
    return {
      order: "descending",
      sort: "new",
      dropdown_sort: [{ title: "new" }, { title: "popular" }]
    };
  },
  methods: {
    orderUpdate: function() {
      if (this.order == "descending") {
        this.order = "ascending";
      } else if (this.order == "ascending") {
        this.order = "descending";
      }
      this.$emit("order", this.order);
    },
    sortbyUpdate: function(newSort) {
      this.sort = newSort;
      this.$emit("sort", this.sort);
    }
  }
};
</script>