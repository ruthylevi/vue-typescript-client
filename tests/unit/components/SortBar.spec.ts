import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import { shallowMount } from '@vue/test-utils';
import SortBar from '@/components/SortBar.vue';

Vue.use(Vuetify);
Vue.use(VueRouter);

const wrapper = shallowMount(SortBar);

describe('SortBar', () => {
  test('Sanity', () => {
    expect(true).toBe(true);
  });

  test('Visible', () => {
    expect(wrapper.isVisible()).toBe(true);
  });

  test('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
