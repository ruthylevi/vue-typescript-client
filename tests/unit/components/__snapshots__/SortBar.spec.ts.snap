// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar Renders correctly 1`] = `
<v-toolbar-stub tag="header" tile="true" dense="true" extensionheight="48" src="">
  <v-menu-stub opendelay="0" closedelay="0" contentclass="" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" origin="top left" transition="v-menu-transition">
    <v-list-stub tag="div" tile="true">
      <v-list-item-stub activeclass="" tag="div">
        <v-list-item-title-stub>new</v-list-item-title-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div">
        <v-list-item-title-stub>popular</v-list-item-title-stub>
      </v-list-item-stub>
    </v-list-stub>
  </v-menu-stub>
  <v-btn-stub tag="button" activeclass="" type="button">
    <v-icon-stub>mdi-sort-descending</v-icon-stub>
  </v-btn-stub>
  <v-spacer-stub></v-spacer-stub>
  <v-spacer-stub></v-spacer-stub>
  <v-spacer-stub></v-spacer-stub>
  <v-btn-toggle-stub value="0" activeclass="v-item--active" mandatory="true">
    <v-btn-stub tag="button" activeclass="" type="button" default="">
      <v-icon-stub>mdi-view-grid</v-icon-stub>
    </v-btn-stub>
    <v-btn-stub tag="button" activeclass="" type="button">
      <v-icon-stub>mdi-view-list</v-icon-stub>
    </v-btn-stub>
  </v-btn-toggle-stub>
</v-toolbar-stub>
`;
